src/test/scala/chiselTests/ModuleChoiceSpec.scala

```diff
@@ -129,3 +129,22 @@ class ModuleChoiceSpec extends ChiselFlatSpec with Utils with MatchesAndOmits {
   }
 
 }
+class AddGroupSpec extends ChiselFlatSpec with Utils with MatchesAndOmits {
+  it should "emit options for a registered group even if there are no consumers" in {
+    class ModuleWithoutChoice extends Module {
+      addGroup(Platform)
+      val out = IO(UInt(8.W))
+      val in = IO(UInt(8.W))
+      out := in
+    }
+
+    val chirrtl = ChiselStage.emitCHIRRTL(new ModuleWithoutChoice, Array("--full-stacktrace"))
+
+    info("CHIRRTL emission looks correct")
+    matchesAndOmits(chirrtl)(
+      "option Platform :",
+      "FPGA",
+      "ASIC"
+    )()
```

Comment:
I landed some of Jack's work to make this checking more strict. Instead of `matchesAndOmits`, can this use the following:

```suggestion
    fileCheckString(chirrtl)(
      """|CHECK: option Platform :
         |CHECK-NEXT: FPGA
         |CHECK-NEXT: ASIC
         |""".stripMargin
    )()
```

This will need to mix `chiselTests.FileCheck` into `AddGroupSpec`.

Reply:
Also, inlining `chirrtl` into the usage site can allow you to use `generateFirrtlAndFileCheck`.